cpp/mrc/src/internal/data_plane/data_plane_resources.cpp

```diff
@@ -363,6 +364,58 @@ std::shared_ptr<ucxx::Request> DataPlaneResources2::am_recv_async(std::shared_pt
     return request;
 }
 
+uint64_t DataPlaneResources2::get_next_object_id()
+{
+    auto object_id = m_next_object_id++;
+    while (m_remote_descriptor_by_id.find(object_id) != m_remote_descriptor_by_id.end())
+    {
+        object_id = m_next_object_id++;
+    }
+
+    return object_id;
+}
+
+uint64_t DataPlaneResources2::register_remote_decriptor(
+    std::shared_ptr<runtime::RemoteDescriptorImpl2> remote_descriptor)
+{
+    auto object_id = get_next_object_id();
+    remote_descriptor->encoded_object().set_object_id(object_id);
+    m_remote_descriptor_by_id[object_id] = remote_descriptor;
+    return object_id;
+}
+
+void DataPlaneResources2::decrement()
+{
+    std::map<std::shared_ptr<ucxx::Endpoint>, std::shared_ptr<ucxx::Request>> probeRequests;
+    for (auto& ep_pair : m_endpoints_by_address)
+    {
+        auto ep     = ep_pair.second;
+        auto worker = ep->getWorker();
+        if (worker->tagProbe(0))
```

Comment:
Is it a requirement to receive have an endpoint for receiving a tagged message? If so, we may need to rethink this a bit to allow receiving messages from endpoints which have not been registered.

Reply:
No, we don't need an endpoint here and we can use the worker instead. As I mentioned previously, I don't think using tags is a good solution and I just wrote it like this to have a working example. I have better idea in mind for this case, let's discuss it offline later today.

Reply:
Resolving since this is outdated